pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/service_proxy_transport.go

```diff
@@ -0,0 +1,203 @@
+package openshiftkubeapiserver
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/api/core/v1"
+	kapierrors "k8s.io/apimachinery/pkg/api/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	listersv1 "k8s.io/client-go/listers/core/v1"
+	restclient "k8s.io/client-go/rest"
+	cache "k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+// serviceCABundleRoundTripper creates a new RoundTripper for each request with serverName and caBundle input into the
+// TLSClientConfig. It is expected that caBundle is kept up to date by the serviceCABundleUpdater controller.
+type serviceCABundleRoundTripper struct {
+	serverName string
+	caBundle   []byte
+	lock       sync.Mutex
+}
+
+func (r *serviceCABundleRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+	newRestConfig := &restclient.Config{
```

Comment:
This step is the one we want to avoid each time. Listing is cheap, building the transport is comparatively expensive